src/variant.rs

```diff
@@ -378,6 +379,90 @@ impl<T: ToVariant> From<T> for Variant {
     }
 }
 
+/// Returns `VariantType` of `Self`.
+pub trait DynamicVariantType {
```

Comment:
Maybe `StaticVariantType` can be completely replaced by this, and instead of returning a `VariantType` you would return a `Cow<'static, VariantTy>`?

Reply:
Sure but then `VariantTy` also needs to be changed to container a ref rather.

Reply:
That's how it already works. There's `&VariantTy` (like `&str`) and `VariantType` (like `String`).

Reply:
I'd keep the name of the new trait as `StaticVariantType` btw, it's still the static signature of some type (it's not dependent on a value, it's fixed at compile time).